src/perl/Permabit/BlockDevice/VDO.pm

```diff
@@ -342,8 +342,10 @@ sub fillIndex {
 ##
 sub activate {
   my ($self) = assertNumArgs(1, @_);
-  $self->SUPER::activate();
   $self->addDeactivationStep(sub { $self->stopVDO(); });
+
+  # Always call SUPER::activate at end to do final initialization.
+  $self->SUPER::activate();
 }
```

Comment:
I reference the non lvm vdo's in the commit message so yeah you're correct that it only affects them. I first noticed a general issue with DoryFastRebuild since that uses Unmanaged. It really stood out to me that the ordering seemed very different from every other BlockDevice type we have which calls super::activate at the very end. We could use predeactivate steps but it seems like a workaround to me. I don't know. I'd be interested to hear Ken's take on this.

Reply:
As a matter of logic that doesn't make any sense to me. Doing pattern-matching on "this is what other functions do" is not a substitute for thinking about why we do this and if it matter that the order changes

In particular, the "add deactivation" step should happen immediately after the thing that causes the deactivation to be necessary. It should never be the first thing in a function.

I just looked through the devices, and the deactivation step might be in the wrong place; it's a question of whether the stopVDO() function needs the chmod to have happened in order to function correctly. If the chmod is not required, it would be clearer to move the "add deactivation step" up into Unmanaged::activate() and Managed::activate(). That's probably a more appropriate fix for this issue. And it'll fix the deactivation order problem, also.

Reply:
I think that change would also mean you could get rid of the stopVDO() function in BlockDevice::VDO,  which would probably remove some confusion.

Reply:
As Bruce pointed out later, super::activate needs to be called after the device has been created so it can do the chmod. So I think this ordering probably does make sense.

Reply:
It is being called after the device is created, as evidenced by the fact that it worked. The issue is that the deactivation step is not attached to the place where the device is being created.

Reply:
Moving it probably makes sense. I'll look into it on monday